source/extensions/http/cache/file_system_http_cache/file_system_http_cache.cc

```diff
@@ -71,11 +81,16 @@ absl::string_view FileSystemHttpCache::name() {
 }
 
 FileSystemHttpCache::FileSystemHttpCache(
-    Singleton::InstanceSharedPtr owner, ConfigProto config,
-    std::shared_ptr<Common::AsyncFiles::AsyncFileManager>&& async_file_manager,
+    Singleton::InstanceSharedPtr owner, CacheEvictionThread& cache_eviction_thread,
+    ConfigProto config, std::shared_ptr<Common::AsyncFiles::AsyncFileManager>&& async_file_manager,
     Stats::Scope& stats_scope)
     : owner_(owner), config_(config), async_file_manager_(async_file_manager),
-      stats_(generateStats(stats_scope, cachePath())) {}
+      stats_(generateStats(stats_scope, cachePath())),
```

Comment:
can we keep some eviction stats? e.g.:
 * how much time is busy vs idle in the eviction thread
 * number of files evicted
 * number of bytes evicted

Reply:
I have included this in DESIGN.md as an unchecked goal and postponed it from this change. We like small PRs, right? :)

Reply:
Done.